### PR TITLE
Dojo Widgets 7 Announcement Blog

### PR DESCRIPTION
**Pull request by ericos**
## Description

Initial draft of the Dojo Widgets 7 announcement blog.

### Code

This PR touches:

- [x] Content
- [ ] Content Pipeline
- [ ] Frontend
- [ ] Infrastructure

### Tests

- [ ] Tests are included?

### CODE REVIEW

File: content/blog/en/version-7-dojo-widgets.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
+---
+title: Announcing Dojo 7 Widgets
+date: 2019-08-28T08:00:00.000Z
+author: Anthony Gubler
```
Comment (agubler) [inline content/blog/en/version-7-dojo-widgets.md:4]: Hey Eric! You are the author!

File: content/blog/en/version-7-dojo-widgets.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
+---
+title: Announcing Dojo 7 Widgets
+date: 2019-08-28T08:00:00.000Z
```
Comment (agubler) [inline content/blog/en/version-7-dojo-widgets.md:3]: Let's set the date for Monday,  `date: 2020-04-20`

File: content/blog/en/version-7-dojo-widgets.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+Providing maximum value to Dojo users means focusing on providing users a strong suite of widgets out of the box. Dojo 7 includes over two dozen new widgets that developers can use to build feature-rich applications even faster. Included are specialized widget variants, data-aware widgets, and widgets for building column layouts.
+
+Dojo 7 includes a key new feature called resources which are a combination of a middleware and resource factory that enables widget authors to build data-aware widgets with a standard, easy to use, and understandable API.
+
+This new powerful feature is at the heart of the data-aware widgets that were built for Dojo 7 and we will be looking for feedback from our users to further refine the widgets in future releases.
+
+The new Card widget provides a simple and consistently styled container for content and actions.
+
+![Card Widget](assets/blog/version-7-dojo-widgets/cardWidget.png)
```
Comment (ericos) [inline content/blog/en/version-7-dojo-widgets.md:29]: When showcasing some of the new widgets, do we want to include images, code, both?

Also, any additional/different widgets we should show off?

**Comment by ericos** [comment]
We can also add a support section at the end.

File: content/blog/en/version-7-dojo-widgets.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+## Expanded Widget Library
+
+Providing maximum value to Dojo users means focusing on providing users a strong suite of widgets out of the box. Dojo 7 includes over two dozen new widgets that developers can use to build feature-rich applications even faster. Included are specialized widget variants, data-aware widgets, and widgets for building column layouts.
+
+Dojo 7 includes a key new feature called resources which are a combination of a middleware and resource factory that enables widget authors to build data-aware widgets with a standard, easy to use, and understandable API.
+
+This new powerful feature is at the heart of the data-aware widgets that were built for Dojo 7 and we will be looking for feedback from our users to further refine the widgets in future releases.
+
+The new Card widget provides a simple and consistently styled container for content and actions.
```
Comment (rorticus) [inline content/blog/en/version-7-dojo-widgets.md:27]: The segue to this paragraph was strange for me. I had to read it a few times to realize that the Card widget doesn't have anything to do with the last two paragraphs. Anyway we can transition this easier?

Comment (ericos) [reply content/blog/en/version-7-dojo-widgets.md]: Agreed. Cleaned this up.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-qk5nzppil.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-hdtc7gogl.now.sh

File: content/blog/en/version-7-dojo-widgets.md [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+			)}
+		</virtual>
+	);
+});
+
+export default App;
+
+```
+
+Check out all of the new Dojo widgets at [https://widgets.dojo.io](https://widgets.dojo.io/)
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:200]:
```suggestion
View the new [Dojo widgets in action](https://widgets.dojo.io/).
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+
+- Expanding the widget library, focusing on widgets that provide the most value to end users.
+- Refactoring all widgets as function-based and in TSX.
+- Adopt the latest recommended Dojo widget authoring patterns.
+- Improve out of the box usability of each widget.
+- Standardize widget property patterns across the library.
+- Improve the theming experience for end users.
+- Expand documentation and widget examples.
+
+## Expanded Widget Library
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:19]:
```suggestion
## Expanded Dojo Widgets library
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+
+Providing maximum value to Dojo users means focusing on providing users a strong suite of widgets out of the box. Version 7 of Dojo Widgets includes over two dozen new widgets that developers can use to build feature-rich applications even faster. Included are specialized widget variants, data-aware widgets, and widgets for building column layouts.
+
+#### New Widgets
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:23]:
```suggestion
#### New Dojo Widgets
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+	);
+});
+
+export default App;
+
+```
+
+Check out all of the new Dojo widgets at [https://widgets.dojo.io](https://widgets.dojo.io/)
+
+## Adopting the Latest Recommended Dojo Widget Authoring Patterns
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:202]:
```suggestion
## Adopting the latest recommended Dojo Widget authoring patterns
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+
+The introduction of function-based widgets fundamentally changed the recommended authoring patterns for Dojo widgets, providing more flexibility and functionality when used in combination with middleware. For simple widgets, the advantages may not be obvious, however, when working on more complex widgets these advantages become more apparent. One of the powerful key features of function-based widgets is being able to support typed children renderers, replacing the current render property pattern. There are several key advantages to using functional children over render properties:
+
+- The Dojo rendering engine deals with the invalidation strategy internally.
+- Typed children provide more flexibility when authoring widgets and most importantly more safety for the end user.
+
+For Dojo 7, all widgets are now function-based, written in TSX, and reflect the latest recommended widget authoring patterns.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:209]:
```suggestion
For Dojo 7, all widgets are now function-based, leverage the TSX syntax, and reflect the latest recommended widget authoring patterns.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+
+## Making Widgets More Usable Out of the Box
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:211]:
```suggestion
## Making Dojo Widgets more usable out-of-the-box
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+
+As part of Dojo 7, all widgets were reviewed for usability issues with an emphasis on improving the developer experience. Property patterns were standardized across widgets and many are now partially controlled, handling their own state. The result is a more consistent and accessible collection of widgets that is usable right out of the box.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:213]:
```suggestion
As part of Dojo version 7, all widgets received and extensive review for usability issues with an emphasis on improving the developer experience. Property patterns were standardized across widgets and many widgets are now partially controlled, handling their own state. The result is a more consistent, usable and accessible collection of widgets.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+
+### Standardization of Widget Property Patterns
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:215]:
```suggestion
### Standardization of Dojo Widget property patterns
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+
+This release brings more consistency to property patterns throughout the widgets library. The result is a reduced learning curve and improved developer experience. Common widget properties are aligned across all widgets, ensuring familiarity for end users.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:217]:
```suggestion
This release brings more consistency to property patterns throughout the Dojo Widgets library. The result is a reduced learning curve and improved developer experience. Common widget properties are now aligned across all widgets, ensuring familiarity for end users.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 14]
```diff
@@ -0,0 +1,307 @@
...
+
+#### value and onValue
+
+The use of `onInput` / `onChange` / etc has been consolidated into a consistent `onValue` callback. All widgets returning a value will do so using this callback. Additionally, callbacks that previously returned a value or key, such as `onChange` or `onBlur` have been removed or updated to return zero parameters.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:221]:
```suggestion
The use of `onInput` / `onChange` / etc has been consolidated into a consistent `onValue` callback. All widgets returning a value leverage this callback. Additionally, callbacks that previously returned a value or key, such as `onChange` or `onBlur` have been removed or updated to return zero parameters. Dojo previously tried to simply align with how HTML worked, but the inconsistency across HTML elements is frustrating and having a consistent pattern across all widgets streamlines cognitive overhead when working with Dojo Widgets.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 15]
```diff
@@ -0,0 +1,307 @@
...
+
+#### mouse / touch events
+
+Mouse and touch events have been standardized to use pointer events, allowing for removal of a large number of callbacks. Rather than `mouseIn` and `mouseOut`, widgets now provide `onOver` and `onOut`.
+
+### Partial Control of Widgets
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:227]:
```suggestion
### Partial control of Dojo Widgets
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 16]
```diff
@@ -0,0 +1,307 @@
...
+
+One goal with Dojo 7 is to make widgets work out of box wherever possible, without extra boilerplate for controlling the widget state. As a result, many widgets have been updated to use the `icache` middleware and now maintain their own state. This is accompanied by a standardized `initialValue` property that can control the `value` but does not need to be set each time the `onValue` callback is called. These widgets can still be optionally controlled using the `value` and `onValue` properties for backward compatibility, but we believe this new pattern provides an improvement to the developer experience.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:229]:
```suggestion
One goal with Dojo 7 is to make widgets work out-of-the-box wherever possible, without extra boilerplate for controlling the widget state. As a result, many widgets have been updated to use the `icache` middleware and now maintain their own state. This is accompanied by a standardized `initialValue` property that can control the `value` but does not need to be set each time the `onValue` callback is called. These widgets can still optionally get controlled using the `value` and `onValue` properties for backward compatibility, but we believe this new pattern provides an improvement to the developer experience.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 17]
```diff
@@ -0,0 +1,307 @@
...
+
+const factory = create({});
+
+export default factory(function Basic({}) {
+	return <Slider min={0} max={100} />;
+});
+```
+
+
+## Improved Theming Experience
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:270]:
```suggestion
## Improved theming experience
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 18]
```diff
@@ -0,0 +1,307 @@
...
+
+A primary goal of the Dojo 7 release is to make widget theming more accessible to end users. To that end, the @dojo/themes repository was deprecated, and themes were moved to @dojo/widgets. This allows developers to start building great looking applications even faster. Additional improvements include the inclusion of a Material theme and support for theme variants.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:272]:
```suggestion
A primary goal of the Dojo 7 release is to make widget theming more accessible to end users. To that end, the `@dojo/themes` repository was deprecated, and themes were moved to `@dojo/widgets`. This allows developers to start building great looking applications even faster. Additional improvements include the inclusion of a Material theme and support for theme variants.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 19]
```diff
@@ -0,0 +1,307 @@
...
+
+**Material theme**
+
+The Dojo 7 release brings a new Material theme to widgets. Beyond providing another modern appearance option for Dojo applications, the Material theme serves as an example of theming best practices. Development of the Material theme also helped validate several patterns in Dojo and helped guide the implementation of new features, such as theme variants.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:276]:
```suggestion
The Dojo 7 release brings a new Material Design theme to Dojo Widgets. Beyond providing another modern appearance option for Dojo applications, the Material theme serves as an example of theming best practices. Development of the Material theme also helped validate several patterns in Dojo and helped guide the implementation of new features such as theme variants.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 20]
```diff
@@ -0,0 +1,307 @@
...
+
+Through the use of css-modules, providing a Material theme for many widgets was as straightforward as composing the appropriate classes from the @material packages.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:278]:
```suggestion
Through the use of css-modules, providing a Material theme for many widgets simply required composing the appropriate classes from the @material packages.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 21]
```diff
@@ -0,0 +1,307 @@
...
+.active {
+	composes: mdc-floating-label--float-above from '@material/textfield/dist/mdc.textfield.css';
+}
+
+.root.secondary {
+	position: relative;
+}
+```
+
+**Theme Variants**
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:297]:
```suggestion
**Theme variants**
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 22]
```diff
@@ -0,0 +1,307 @@
...
+
+Themes in Dojo 7 now support 'variants'. This allows developers to easily create variations of a theme with minimal duplication. By loading new css-variables, developers can include light, dark, high-contrast, or any number of variants within a theme.
+
+## Revamped Documentation and Examples
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:301]:
```suggestion
## Revamped Dojo Widgets documentation and examples
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 23]
```diff
@@ -0,0 +1,307 @@
...
+
+This release includes improved widget documentation and examples. The widget documentation site at [widgets.dojo.io](https://widgets.dojo.io) uses the new [@dojo/parade](https://github.com/dojo/parade) package to provide a consistent and up-to-date reference guide. Multiple examples are provided for each widget, and Dojo Parade provides auto-documentation of widget properties and theme class hooks. This new documentation tool is built with Dojo and offers a great tool for building widget documentation and examples from within your application or widget library.
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:303]:
```suggestion
This release includes improved Dojo Widgets documentation and examples. The [Dojo Widgets documentation site](https://widgets.dojo.io) uses the new [@dojo/parade](https://github.com/dojo/parade) package to provide a consistent and up-to-date reference guide. Multiple examples are provided for each widget, and Dojo Parade provides auto-documentation of widget properties and theme class hooks! This new documentation tool is built with Dojo and provides a great tool for building widget documentation and examples from within your application or widget library.
```

File: content/blog/en/version-7-dojo-widgets.md [hunk 24]
```diff
@@ -0,0 +1,307 @@
...
+
+## Migration
+
+All breaking changes in this release were carefully considered and focused on creating the best developer experience. For information on what has changed in Dojo Widgets, please see the [v7 migration guide](https://github.com/dojo/widgets/blob/master/docs/V7-Migration-Guide.md).
```
Comment (dylans) [inline content/blog/en/version-7-dojo-widgets.md:307]:
```suggestion
All breaking changes in this release were carefully considered and focused on creating the best developer experience. For information on what has changed in Dojo Widgets, please see the [Dojo Widgets version 7 migration guide](https://github.com/dojo/widgets/blob/master/docs/V7-Migration-Guide.md).
```

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-n7mkpxhif.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-e9s0t6ndk.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-f14bpa60h.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-1kw6tk491.now.sh

File: content/blog/en/version-7-dojo-widgets.md [hunk 25]
```diff
@@ -0,0 +1,307 @@
...
+		</div>
+	);
+});
+```
+
+![Card Widget](assets/blog/version-7-dojo-widgets/cardWidget.png)
+
+The [PasswordInput widget](https://widgets.dojo.io/#widget/password-input/overview) is one of several new specialized widget variants, automatically handling validation state and messaging internally saving development time and improving productivity.
+
+> src/MyPasswordInput.tsx
```
Comment (tomdye) [inline content/blog/en/version-7-dojo-widgets.md:73]: should this not be `App.tsx`?

File: content/blog/en/version-7-dojo-widgets.md [hunk 26]
```diff
@@ -0,0 +1,307 @@
+---
+title: Announcing Dojo 7 Widgets
+date: 2020-04-20T08:00:00.000Z
+author: Eric Osmundson
+---
+
+Since the last Dojo release, we have been expanding the features and improving the developer experience of the [Dojo Widgets](https://github.com/dojo/widgets) library. With new advances in Dojo features, functionality, and recommended best practices, Dojo version 7 provides substantial improvements to make it easier for developers to build efficient and modern web applications.
+
```
Comment (agubler) [inline content/blog/en/version-7-dojo-widgets.md:8]: Could you add `<!-- more -->` here please? And I have requested a featured image for the blog that can go in once we have it

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-jads3c21l.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-9veu7qmq3.now.sh

File: content/blog/en/version-7-dojo-widgets.md (hunk 26 above)
Comment (ericos) [reply content/blog/en/version-7-dojo-widgets.md:8]: Images updated, including featured image.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-7t2xa3ox9.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-87kcwdse7.now.sh

File: content/blog/en/version-7-dojo-widgets.md [hunk 27]
```diff
@@ -0,0 +1,307 @@
...
+- Improve the theming experience for end users.
+- Expand documentation and widget examples.
+
+## Expanded Widget Library
+
+Providing maximum value to Dojo users means focusing on providing users a strong suite of widgets out of the box. Version 7 of Dojo Widgets includes over two dozen new widgets that developers can use to build feature-rich applications even faster. Included are specialized widget variants, data-aware widgets, and widgets for building column layouts.
+
+#### New Widgets
+
+The new Card widget provides a simple and consistently styled container for content and actions. It uses the new child renderer pattern, accepting content for different sections and rendering them in the appropriate location with appropriate styling.
```
[the remark this answers is not in the document]
Comment (ericos) [reply content/blog/en/version-7-dojo-widgets.md]: Linked.

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-7u27d7u6r.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-l0ohp0oke.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-52jy1cnwd.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-owx23e9tk.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* next-dojo-io: https://next-dojo-io-ox41m0kma.now.sh
* zh-cn-next-dojo-io: https://zh-cn-next-dojo-io-f0o4wwo4i.now.sh

**Comment by agubler** [comment]
Thanks @ericos, really appreciate getting this finished. I've pulled this into the other blog PR so it can be merged together.

**Pull request closed without merging**